src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4393,3 +4393,181 @@ void C2_MacroAssembler::vector_maskall_operation32(KRegister dst, Register src,
   kunpckdql(dst, tmp, tmp);
 }
 #endif
+
+void C2_MacroAssembler::vector_reverse_bit_avx(BasicType bt, XMMRegister dst, XMMRegister src, XMMRegister xtmp1,
```

Comment:
You can do a this bit reverse using lookup table on each nibble and oring the results, the pseudocode would look something like this

    lut = broadcasti128(0b0000, 0b1000, 0b0100, 0b1100, 0b0010, 0b1010, 0b0110, 0b1110, 0b0001, ...)
    mask = pbroadcastd(0x0f0f0f0f)
    hi = pand(src, mask)
    hi = pshufb(lut, hi)
    hi = pslld(hi, 4)
    lo = psrld(src, 4)
    lo = pand(lo, mask)
    lo = pshufb(lut, lo)
    dst = por(lo, hi)

Reply:
Yes, this can be tried out. I guess its inspired from LUT based revbits algorithm from Hacker's delight section 7-1.

Reply:
@merykitty , your comments addressed.